From a99ce949d07af78e5346ab31d3dc01539b15e0bf Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Tue, 28 Apr 2020 23:10:00 +0900
Subject: [PATCH] VIM3_GPIOD_OPEN_DRAIN_hdmi_fix

khadas dts have GPIOD_OPEN_DRAIN which cant parsed properly
and cant power video system

---
 drivers/gpio/gpio-uclass.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpio/gpio-uclass.c b/drivers/gpio/gpio-uclass.c
index 757ab710..ddfcd12d 100644
--- a/drivers/gpio/gpio-uclass.c
+++ b/drivers/gpio/gpio-uclass.c
@@ -524,6 +524,15 @@ int dm_gpio_set_value(const struct gpio_desc *desc, int value)
 	if (ret)
 		return ret;

+	if (desc->flags & GPIOD_OPEN_DRAIN) {
+		if (value)
+		    gpio_get_ops(desc->dev)->direction_input(desc->dev, desc->offset);
+		else
+		    gpio_get_ops(desc->dev)->direction_output(desc->dev, desc->offset,0);
+
+		return 0;
+       }
+
 	if (desc->flags & GPIOD_ACTIVE_LOW)
 		value = !value;
 	gpio_get_ops(desc->dev)->set_value(desc->dev, desc->offset, value);
--
2.17.1
